src/MainWindow.vala

```diff
@@ -847,4 +882,96 @@ public class PantheonCalculator.MainWindow : Gtk.ApplicationWindow {
             Signal.stop_emission_by_name ((void*) entry.get_delegate (), "insert-text");
         }
     }
+
+    private bool on_key_press (Gtk.EventControllerKey controller, uint keyval, uint keycode, Gdk.ModifierType mod_state) {
+        event_controller.forward (entry.get_delegate ());
```

Comment:
The controller is passed as an argument, you can reuse it instead of storing event controller in window.

Reply:
?